File: Projeto_NFe/Server/Projeto_NFe.Infra.PDF/Features/Issuers/IssuerModel.cs
using Projeto_NFe.Domain.Features.Issuers;
using Projeto_NFe.Infra.PDF.Features.Addresses;

namespace Projeto_NFe.Infra.PDF.Features.Issuers
{
    public class IssuerModel
    {
        public int Id { get; set; }
        public string BusinessName { get; set; }
        public string CorporateName { get; set; }
        public string Cnpj { get; set; }
        public string StateRegistration { get; set; }
        public string MunicipalRegistration { get; set; }
        public AddressModel Address { get; set; }

        public void Create(Issuer issuer)
        {
            Cnpj = issuer.Cnpj;
            Address = new AddressModel();
            Address.Neighborhood = issuer.Address.Neighborhood;
            Address.State = issuer.Address.State;
            Address.StreetName = issuer.Address.StreetName;
            Address.City = issuer.Address.City;
            Address.Number = issuer.Address.Number;
            Address.Country = issuer.Address.Country;
            StateRegistration = issuer.StateRegistration;
            MunicipalRegistration = issuer.MunicipalRegistration;
            BusinessName = issuer.BusinessName;
            CorporateName = issuer.CorporateName;
        }
    }
}

File: Projeto_NFe/Client/src/app/features/issuer/shared/issuer.model.ts
import { Address, AddressCommand } from './../../address/address.model';
export class Issuer {
    public id: number;
    public businessName: string;
    public corporateName: string;
    public cnpj: string;
    public stateRegistration: string;
    public municipalRegistration: string;
    public streetName: string;
    public number: string;
    public neighborhood: string;
    public city: string;
    public state: string;
    public country: string;
}

export class IssuerRegisterCommand {
    public businessName: string;
    public corporateName: string;
    public cnpj: string;
    public stateRegistration: string;
    public municipalRegistration: string;
    public address: AddressCommand;

    constructor(issuer: Issuer, address: AddressCommand) {
        this.businessName = issuer.businessName;
        this.corporateName = issuer.corporateName;
        this.cnpj = issuer.cnpj;
        this.stateRegistration = issuer.stateRegistration;
        this.municipalRegistration = issuer.municipalRegistration;
        this.address = address;
    }
}

export class IssuerUpdateCommand {
    public id: number;
    public businessName: string;
    public corporateName: string;
    public cnpj: string;
    public stateRegistration: string;
    public municipalRegistration: string;
    public address: AddressCommand;

    constructor(issuer: Issuer, address: AddressCommand, id: number) {
        this.id = id;
        this.businessName = issuer.businessName;
        this.corporateName = issuer.corporateName;
        this.cnpj = issuer.cnpj;
        this.stateRegistration = issuer.stateRegistration;
        this.municipalRegistration = issuer.municipalRegistration;
        this.address = address;
    }
}

export class IssuerRemoveCommand {
    public issuersId: number[] = [];

    constructor(selectedEntities: any) {
        this.issuersId = selectedEntities.map((issuer: Issuer) => issuer.id);
    }
}
